File: RepoTransBench/source_projects/Python/requests_requests-oauthlib/public_tests/test_public_oauth2_auth.py
import unittest

from requests_oauthlib.oauth2_auth import OAuth2

class TestPublicOAuth2(unittest.TestCase):
    def test_public_oauth2_auth_header_diff_data(self):
        client_id = "public_client_id"
        token_data = {
            "access_token": "tok_9876543210abc",
            "token_type": "Bearer",
            "expires_in": 600,
        }
        oauth = OAuth2(client_id, token=token_data)
        class Req:
            def __init__(self):
                self.headers = {}
        req = Req()
        r2 = oauth(req)
        self.assertIn("Authorization", r2.headers)
        self.assertTrue(r2.headers["Authorization"].startswith("Bearer "))
        self.assertIn(token_data["access_token"], r2.headers["Authorization"])

    def test_public_oauth2_auth_repr_diff(self):
        oauth = OAuth2("public_id", token=None)
        rep = repr(oauth)
        self.assertIn("public_id", rep)
        self.assertTrue(rep.startswith("OAuth2("))

File: RepoTransBench/target_projects/Python/C++/requests_requests-oauthlib/public_tests/test_public_oauth2_auth.cpp
#include <gtest/gtest.h>
#include "requests_oauthlib/OAuth2.h"

TEST(TestPublicOAuth2, PublicOAuth2AuthHeaderDiffData) {
    std::map<std::string, std::string> token = {
        {"access_token", "tok_9876543210abc"},
        {"token_type", "Bearer"},
        {"expires_in", "600"}
    };
    OAuth2 oauth("public_client_id", token);
    std::string req = "GET /";
    std::string out = oauth(req);
    EXPECT_TRUE(out.find("Bearer tok_9876543210abc") != std::string::npos);
}

TEST(TestPublicOAuth2, PublicOAuth2AuthReprDiff) {
    OAuth2 oauth("public_id");
    std::string rep = oauth.repr();
    EXPECT_TRUE(rep.find("public_id") != std::string::npos);
    EXPECT_TRUE(rep.find("OAuth2(") == 0);
}